test/e2e/utils/cnpspecbuilder.go

```diff
@@ -162,7 +162,8 @@ func (b *ClusterNetworkPolicySpecBuilder) AddIngress(protoc v1.Protocol,
 	for _, at := range ruleAppliedToSpecs {
 		appliedTos = append(appliedTos, b.GetAppliedToPeer(at.PodSelector, at.NSSelector, at.PodSelectorMatchExp, at.NSSelectorMatchExp, at.Group))
 	}
-	var policyPeer []crdv1alpha1.NetworkPolicyPeer
+	// An empty From/To in ACNP rules evaluates to match all addresses.
+	policyPeer := make([]crdv1alpha1.NetworkPolicyPeer, 0)
```

Comment:
I noticed you updated this line to fix nil slice unmarshall, could you point out what code it would affect? Could the issue be fixed by adding `omitempty` to `From` and `To`?

Reply:
It only affects when ACNP is constructed programmatically, i.e from this file. We should use a 0 length slice for policy peers instead of a `nil` slice. It does not affect user side, as empty `From/To` in a yaml file will be interpreted correctly by the API server